src/transaction/mod.rs

```diff
@@ -10,20 +10,363 @@
 //!
 
 pub use self::client::{Client, Connect};
-pub use self::requests::{BatchGet, Commit, Delete, Get, LockKeys, Rollback, Scanner, Set};
-pub use self::transaction::{IsolationLevel, Snapshot, Transaction, TxnInfo};
-pub use super::rpc::Timestamp;
-
+use self::requests::{BatchGet, Commit, Delete, Get, LockKeys, Rollback, Scanner, Set};
 use crate::{Key, Value};
+use derive_new::new;
+use kvproto::kvrpcpb;
+use std::ops::RangeBounds;
 
 mod client;
-mod requests;
-#[allow(clippy::module_inception)]
-mod transaction;
+pub mod requests;
+
+#[derive(Clone, Eq, PartialEq, Debug)]
+pub struct Timestamp {
+    pub physical: i64,
+    pub logical: i64,
+}
 
 pub enum Mutation {
     Put(Key, Value),
     Del(Key),
     Lock(Key),
     Rollback(Key),
 }
+
```

Comment:
I feel good about moving them to the root of the transaction mod. But there are already some modifications. You have to resolve the conflicts. Really sorry.

Reply:
no problem!